libkbfs/root_metadata.go

```diff
@@ -285,6 +285,25 @@ func (md *RootMetadata) MakeSuccessor(
 	return newMd, nil
 }
 
+// MakeSuccessorWithNewHandle does the same thing as MakeSuccessor,
+// plus it changes the handle.  (The caller is responsible for
+// ensuring that the handle change is valid.)
+func (md *RootMetadata) MakeSuccessorWithNewHandle(
+	ctx context.Context, newHandle *TlfHandle, latestMDVer kbfsmd.MetadataVer,
+	codec kbfscodec.Codec, keyManager KeyManager, merkleGetter merkleRootGetter,
+	teamKeyer teamKeysGetter, mdID kbfsmd.ID, isWriter bool) (
+	*RootMetadata, error) {
+	mdCopy, err := md.deepCopy(codec)
+	if err != nil {
+		return nil, err
+	}
+
+	mdCopy.tlfHandle = newHandle.deepCopy()
```

Comment:
On the way home I realized this isn't enough, I also need to fix up `md.bareMd` based on this handle.  The new code should be pretty isolated and probably reviewable on its own, so feel free to review this as is first and then the new commits when I finish them tomorrow, or feel free to wait til it's all ready.

Reply:
Ok I fixed it, ready to review in full again.